monkey/common/agent_events/file_encrypted_event.py

```diff
@@ -0,0 +1,70 @@
+from pathlib import PurePath, PurePosixPath, PureWindowsPath
+from typing import Any, Dict, Mapping, Tuple
+
+from pydantic import Field, validator
+
+from common import OperatingSystem
+
+from . import AbstractAgentEvent
+
+
+def _serialize_pure_path(pure_path: PurePath) -> Dict[str, str]:
+    # We generate data on either Windows or Linux and that data is processed
+    # on either OS which we can't sure where, so be need to serialize it as
+    # dictionary containing the path as string and the operating system
+    # which depends of the type of the path
+    serialized_os = (
+        OperatingSystem.LINUX.value
+        if isinstance(pure_path, PurePosixPath)
+        else OperatingSystem.WINDOWS.value
+    )
+    return {"path": str(pure_path), "os": serialized_os}
+
+
+class FileEncryptedEvent(AbstractAgentEvent):
+    """
+    An event that occurs when the Agent encrypts a file
+
+    Attributes:
+        :param file_path: Path of the encrypted file
+        :param success: Status of the file encryption
+        :param error_message: Message if an error occurs during file encryption
+    """
+
+    file_path: PurePath
```

Comment:
I would've just used a string. If the users of the event need `PurePath` so desperatelly, they can fetch the OS from the machine.